src/webgl/p5.Geometry.js

```diff
@@ -147,6 +156,69 @@ p5.Geometry.prototype.averagePoleNormals = function() {
 };
 
 /**
+ * Create a 2D array for establishing stroke connections
+ * @return {p5.Geometry}
+ */
+p5.Geometry.prototype._makeTriangleEdges = function() {
+  if (Array.isArray(this.strokeIndices)) {
+    for (var i=0, max=this.strokeIndices.length; i<max; i++) {
+      this.edges.push(this.strokeIndices[i]);
+    }
+  }
+  else {
+    for (var j=0, amt=this.faces.length; j<amt; j++) {
+      this._addEdge(this.faces[j][0], this.faces[j][1]);
+      this._addEdge(this.faces[j][1], this.faces[j][2]);
+      this._addEdge(this.faces[j][2], this.faces[j][0]);
+    }
+  }
+  return this;
+};
+
+p5.Geometry.prototype._addEdge = function(i, j) {
+  var edge = [i, j];
+  this.edges.push(edge);
+};
+
+/**
+ * Create 4 vertices for each stroke line, two at the beginning position
+ * and two at the end position. These vertices are displaced relative to
+ * that line's normal on the GPU
+ * @return {p5.Geometry}
+ */
+p5.Geometry.prototype._edgesToVertices = function() {
+  this.lineVertices = [];
+  var vertices = this.lineVertices;
+
+  for(var i = 0, max = this.edges.length; i < max; i++)
+  {
+    var begin = this.vertices[this.edges[i][0]];
+    var end = this.vertices[this.edges[i][1]];
+    var dir = end.copy().sub(begin).normalize();
+    var a = begin,
+        b = begin,
+        c = end,
+        d = end;
+    var dirAdd = dir.array();
+    var dirSub = dir.array();
+    // below is used to displace the pair of vertices at beginning and end
+    // in opposite directions
+    dirAdd.push(1);
+    dirSub.push(-1);
+    this.lineNormals.push(dirAdd,dirSub,dirAdd,dirAdd,dirSub,dirSub);
+    _store([a, b, c, c, b, d]);
+  }
+
+  function _store(verts) {
+    for (var i = 0, max = verts.length; i < max; i += 1) {
```

Comment:
`for (var i = 0; i < verts.length; i++) {`

Reply:
actually, rather than a for loop, I think this is a good place to use the array.map function:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map